perun-dispatcher/src/main/java/cz/metacentrum/perun/dispatcher/scheduling/impl/TaskSchedulerImpl.java

```diff
@@ -246,14 +289,23 @@ public void scheduleTask(Task task) {
 					DependencyScope dependencyScope = dependencyPair.getRight();
 					Task dependencyServiceTask = schedulingPool.getTask(dependency, facility);
 					if (dependencyServiceTask == null) {
-						// Dependency being NULL is equivalent to being in NONE
-						// state.
-						log.info("   Last Task [dependency:"
-								+ dependency.getId() + ", facility:"
-								+ facility.getId()
-								+ "] was NULL, we are gonna propagate.");
-						scheduleItAndWait(dependency, facility, execService,
-								dispatcherQueue, time);
+						if(abortTask) {
+							log.info("   Task {} is going to be aborted, the dependency exec service {} will not be scheduled now.",
+									task.getId(), dependency.getId());
+						} else {
+							// Dependency being NULL is equivalent to being in NONE
+							// state.
+							log.info("   Last Task [dependency:"
+									+ dependency.getId() + ", facility:"
+									+ facility.getId()
+									+ "] was NULL, we are gonna propagate.");
+							if(scheduleItAndWait(dependency, facility, execService,
+									dispatcherQueue, time)) {
+								// task sucessfully scheduled, nothing to do
+							} else {
+								// TODO: task aborted - maybe set this one to error?
```

Comment:
I believe something should be done in else branch (e.g. aborting SEND Task too). If above code causes GEN Task to be aborted, then it's removed from pool completely. So when we goes through pool once again, SEND is ``proceed=true`` (not aborted until here). Here we are ok not to start (create new GEN task), but this whole part sets ``proceed=false`` at the end => we won't reach line 433 where we actually abort the SEND Task. So SEND Task would stay in a pool as PLANNED or NONE? forever.

Above case is when we delete Facility or Service is not assigned. If we delete Service, it should work since GEN-SEND dependencies will be empty and we won't set ``proceed=false``.